src/assets/scripts/client/engine/TimeKeeper.js

```diff
@@ -85,33 +169,142 @@ class TimeKeeper {
+    updateTimewarp(nextTimewarp) {
+        this._timewarp = nextTimewarp;
+    }
+
+    /**
+     *
+     *
+     */
+    setDeltaTimeBeforeFutureTrackCalculation() {
+        this._futureTrackDeltaTimeCache = this._frameDeltaTime;
+        this._frameDeltaTime = 5;
+    }
+
+    /**
+     *
+     *
+     */
+    setDeltaTimeAfterFutureTrackCalculation() {
+        this._frameDeltaTime = this._futureTrackDeltaTimeCache;
+        this._futureTrackDeltaTimeCache = -1;
+    }
+
+    /**
+     *
+     *
+     * @for TimeKeeper
+     * @method shouldUpdate
+     */
+    shouldUpdate() {
+        return this._elapsedFrameCount % this._frameStep === 0;
+    }
+
+    /**
+     *
+     *
+     */
+    togglePause() {
+        this._isPaused = !this._isPaused;
+    }
+
+    /**
+     *
+     *
+     * Should be called at the end of each update cycle by the `AppController`
+     * Calling this method signifies the end of a frame
+     *
+     * @for TimeKeeper
+     * @method update
+     */
+    update() {
+        if (this._futureTrackDeltaTimeCache !== -1) {
+            return;
+        }
+
         const currentTime = this.gameTime;
-        const elapsed = currentTime - this.frameStartTime;
 
-        this.frames += 1;
+        this._incrementFrame();
+        this._calculateNextDeltaTime(currentTime);
+        this._calculateFrameStep();
+    }
+
+    /**
+     * Move to the next frame
+     *
+     * @for TimeKeeper
+     * @method incrementFrame
+     * @private
+     */
+    _incrementFrame() {
+        this._elapsedFrameCount += 1;
+    }
+
+    /**
+     * Caclulate the difference (delta) between the `#currentTime`
+     * and `#_lastFrameTime`.
+     *
+     * This value will be used throughout to app to determine how
+     * much time has passed, thus allowing us to know how much to
+     * move elements.
+     *
+     * @for TimeKeeper
+     * @method _calculateNextDelatTime
+     * @param currentTime {Date}  current date string (in ms)
+     * @private
+     */
+    _calculateNextDeltaTime(currentTime) {
+        const frameDelay = 1;
+        const elapsed = currentTime - this._frameStartTime;
 
         if (elapsed > frameDelay) {
-            this.frameStartTime = currentTime;
+            this._frameStartTime = currentTime;
         }
 
-        this.frameDeltaTime = currentTime - this.lastFrameTime;
-        this.lastFrameTime = currentTime;
+        this._frameDeltaTime = currentTime - this._lastFrameTime;
+        this._lastFrameTime = currentTime;
+        this._accumulatedDeltaTime += this.getDeltaTimeForGameStateAndTimewarp();
+    }
+
+    /**
+     *
+     *
+     * @for CanvasController
+     * @method _calculateFrameStep
+     * @private
+     */
+    _calculateFrameStep() {
+        // TODO: is this even correct? the order of range2 values looks backwards
+        // FIXME: what do the magic numbers mean?
+        this._frameStep = Math.round(extrapolate_range_clamp(1, this._timewarp, 10, 30, 1));
```

Comment:
_Perhaps_ this is in regard to how often we draw a new frame? Here's the logic as I follow it:
```
 1x --> 30 units
 5x --> 15 units
10x -->  1 unit
```

I _feel_ like the most logical `unit` is time related, and probably a number of frames until something else happens. 😕 

Perhaps the fact that this doesn't accommodate timewarps over 10x is why the projections are having issues? Then again, I don't know what the issue was, so this could easily be completely irrelevant to that. Just seems like since it's clamped, when using timewarp above 10 a whole bunch of things might break.